src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -596,6 +596,20 @@ internal override CSharpAttributeData EarlyDecodeWellKnownAttribute(ref EarlyDec
                 return null;
             }
 
+            if (CSharpAttributeData.IsTargetEarlyAttribute(arguments.AttributeType, arguments.AttributeSyntax, AttributeDescription.AsyncMethodBuilderAttribute))
+            {
+                boundAttribute = arguments.Binder.GetAttribute(arguments.AttributeSyntax, arguments.AttributeType, out hasAnyDiagnostics);
+                if (!boundAttribute.HasErrors)
+                {
+                    var typeData = arguments.GetOrCreateData<CommonTypeEarlyWellKnownAttributeData>();
+                    typeData.AsyncMethodBuilderTarget = boundAttribute.ConstructorArguments.FirstOrDefault();
+
+                    // don't return the attribute. We need to bind it again later for full diagnostic reporting
```

Comment:
>// don't return the attribute. We need to bind it again later for full diagnostic reporting [](start = 20, length = 91)

What scenario fails if we return the attribute here?

Reply:
The type referenced in the attribute can be marked obsolete. We have no way to report that during early binding. I tried binding without obsolete checks, then later checking for it, but we've lost the context and so can't easily report the diagnostic location (I could probably store that info in the early data as well, but it started seeming very hacky.

Reply:
Thanks. Please add a test that hits that if there's not an existing test.

Reply:
Added a test for obsolete